cleanupWorker/worker/worker.go

```diff
@@ -0,0 +1,63 @@
+package worker
+
+import (
+	goContext "context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/slovak-egov/einvoice/apiserver/storage"
+	"github.com/slovak-egov/einvoice/cleanupWorker/config"
+	"github.com/slovak-egov/einvoice/cleanupWorker/db"
+	"github.com/slovak-egov/einvoice/pkg/context"
+)
+
+type Worker struct {
+	config  *config.Configuration
+	db      *db.Connector
+	storage *storage.LocalStorage
+}
+
+func New() *Worker {
+	workerConfig := config.New()
+	return &Worker{
+		workerConfig,
+		db.NewConnector(workerConfig.Db),
+		storage.New(workerConfig.LocalStorageBasePath),
+	}
+}
+
+func (w *Worker) Run() {
+	log.Info("worker.started")
+
+	for range time.Tick(w.config.CronInterval) {
+		w.TestInvoicesCleanupJob()
+	}
+}
+
+func (w *Worker) CloseResources() {
+	w.db.Close()
+}
+
+func (w *Worker) TestInvoicesCleanupJob() {
+	ctx := goContext.Background()
+
+	err := w.db.RunInTransaction(ctx, func(ctx goContext.Context) error {
```

Comment:
I would not use transaction.
In this case it can happen deleting from storage failed and so does transaction so invoices are still in DB and our data is broken

Without transaction in worst case we have still in storage some additional data, but DB is reference and it is consistent. 
We also have error log which invoice deleting failed (from storage) and we can fix it manually, but application works as expected all the time